packages/backend/src/payment-method/ilp/connector/core/test/middleware/account-middleware.test.ts

```diff
@@ -120,11 +120,32 @@ describe('Account Middleware', () => {
         rawPrepare: Buffer.alloc(0) // ignored
       }
     })
-    await expect(middleware(ctx, next)).rejects.toThrowError(
+    await expect(middleware(ctx, next)).rejects.toThrow(
       'destination account is disabled'
     )
   })
 
+  test('return an error when the destination account unknown', async () => {
+    const middleware = createAccountMiddleware(ADDRESS)
+    const next = jest.fn()
+    const ctx = createILPContext({
+      state: {
+        incomingAccount,
+        streamDestination: 'no-destination'
+      },
+      services: rafikiServices,
+      request: {
+        prepare: new ZeroCopyIlpPrepare(
+          IlpPrepareFactory.build({ destination: 'test.123' })
+        ),
+        rawPrepare: Buffer.alloc(0) // ignored
+      }
+    })
+    await expect(middleware(ctx, next)).rejects.toThrow(
+      'unknown destination account'
+    )
```

Comment:
The task stated we should check for an `UnreachableError` specifically. I wondered if that was incidental or a core part of the task to ensure we're surfacing errors from [the ilp spec](https://github.com/interledger/rfcs/blob/340a9226f27384c2439a8de7607b76a9333960f8/0023-bilateral-transfer-protocol/0023-bilateral-transfer-protocol.md?plain=1#L286) or something? Probably couldn't hurt to include either way I guess.

```ts
    await expect(middleware(ctx, next)).rejects.toThrow(
      Errors.UnreachableError
    )
```